### PR TITLE
Fix C compiler lookup for relocatable GHC

### PR DESCRIPTION
**Pull request by KaneTW**
This fixes build on Windows.

Additionally, add some rudimentary Windows CI via Appveyor and bump to 0.28.4

### CODE REVIEW

File: src/C2HS/Gen/Bind.hs [hunk 1]
```diff
@@ -3084,8 +3086,21 @@ cCompiler = unsafePerformIO $ do
...
       let vals = read stdout :: [(String, String)]
-      case Prelude.lookup "C compiler command" vals of
-        Nothing -> error "Failed to determine C compiler from 'ghc --info'!"
-        Just cc -> do
-          writeIORef cCompilerRef $ Just cc
-          return cc
+      case (Prelude.lookup "C compiler command" vals, Prelude.lookup "LibDir" vals) of
+        (Just cc, Just topDir) -> do
+          -- ensure that $topdir is expanded
+          let mungedCc = mugnePath topDir cc
```
Comment (deech) [inline src/C2HS/Gen/Bind.hs]: s/`mugnePath`/`mungePath`? :)

Comment (KaneTW) [reply src/C2HS/Gen/Bind.hs]: Yeah, the new version is up in a moment. I ran into a few issue with spaces.

**Comment by KaneTW** [comment]
New version is up. Now handles spaces in names correctly, and adds ".exe" on windows when removing the executable (might want to change it to a temporary directory for easier cleanup?)

**Comment by deech** [comment]
Thank you very much for all this work! Would you mind squashing the commits? I'll review it this weekend.

**Comment by KaneTW** [comment]
Yeah, I'll do it in a sec.

**Comment by deech** [comment]
Wow, I didn't get a notification that you had squashed the PR. Merging now. Will cut a new release in the next couple of days. Thanks!

**Pull request merged**